runtime/sema/typeannotationstate.go

```diff
@@ -27,4 +27,5 @@ const (
 	TypeAnnotationStateValid
 	TypeAnnotationStateInvalidResourceAnnotation
 	TypeAnnotationStateMissingResourceAnnotation
+	TypeAnnotationStateDirectAttachmentTypeAnnotation
```

Comment:
Why is this needed and what does it mean?

Reply:
This is used in `checkTypeAnnotation` to check that naked attachment annotations are rejected statically. The idea is similar to the check that requires that resources have an `@` type.

Reply:
Oh, I see! Could you please add the explanation and maybe an example as a comment to this new case? While at it, could you please do the same for the existing cases? 🙏 

For example: "Resource types must have a resource annotation (@). An annotation is invalid if the annotation is used for a non-resource type, e.g. `@Int`"